### PR TITLE
Use `SameSite` enum from Jakarta

### LINKED ISSUE #26205: Use `SameSite` enum from Jakarta
We're using a custom enum called [`SameSiteAttributeValue`](https://github.com/keycloak/keycloak/blob/fe39d1b5ebba9d489c330f61d7e2d4d95a695d5f/common/src/main/java/org/keycloak/common/util/ServerCookie.java#L35-L47) to control the [`SameSite`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#samesitesamesite-value) attribute on cookies. However, Jakarta already has an enum for this that should be used instead.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #26205

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]
Nice! I also checked to make sure `samesite` value is case-insensitive and it seems to be the case.

Perhaps a follow-up is to check if we can remove more code and rely on the underlying JAX-RS stack. For instance, leverage `NewCookieHeaderDelegate` and standard `RuntimeDelegate` from ResteasyReactive/JAX-RS. But for now, these changes are more than a good start and not so risky, IMO.

@ahus1 @mposolda wdyt?

**Comment by jonkoops** [comment]
> Perhaps a follow-up is to check if we can remove more code and rely on the underlying JAX-RS stack.

Yes, that is exactly what I am working on, figured this would be a good and small place to start since I am still a novice at the Java stuff.

File: common/pom.xml [hunk 1]
```diff
@@ -62,6 +62,10 @@
             <groupId>jakarta.activation</groupId>
             <artifactId>jakarta.activation-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
```
Comment (mposolda) [inline common/pom.xml:67]: Minor: I think this change means that we can probably cleanup this `jakarta.ws.rs-api` dependency from some other files, which depend on `keycloak-common`, which is pretty much everything. For example `services/pom.xml` .

When I currently run:
```
cd services
mvn dependency:tree
```
the output contains:
```
jakarta.ws.rs:jakarta.ws.rs-api:jar:3.1.0:compile
```
I think that with your change to `common/pom.xml` , the dependency can be removed from the `services/pom.xml` and the `mvn dependency:tree` would still show the `jakarta.ws.rs:jakarta.ws.rs-api:jar:3.1.0:compile` (due the services have transitive dependency on `jakarta.ws.rs-api` through `keycloak-common`. Maybe some other places can be cleaned as well in relation to this. WDYT?

**Review by mposolda — COMMENTED** [review]
Added one minor comment regarding dependency (not 100% sure it should be addressed in a way I proposed, but some cleanup might be perhaps good).

**Review by ahus1 — CHANGES_REQUESTED** [review]
Adding the dependency to the `common` module will add this dependency also to several adapters, for example in the Servlet adapter and the Tomcat adapter. This might be unexpected to users, and unintended (not sure if all our adapter are on Jakarta EE yet).

If we want to go the route of JAX-RS, I suggest to not use the single parameter, but instead call the "pure" JAX-RS functionality to set cookies any Keycloak server code, which would be a different refactoring.

File: common/pom.xml (hunk 1 above)
Comment (jonkoops) [reply common/pom.xml:67]: I've removed `jakarta.ws.rs-api` everywhere it was already a transitive dependency, through `common`, or otherwise.

**Comment by jonkoops** [comment]
> If we want to go the route of JAX-RS, I suggest to not use the single parameter, but instead call the "pure" JAX-RS functionality to set cookies any Keycloak server code, which would be a different refactoring.

Yeah, this is essentially what I am working towards. I figured this would be a good starting point into that direction.

**Comment by jonkoops** [comment]
> Adding the dependency to the `common` module will add this dependency also to several adapters, for example in the Servlet adapter and the Tomcat adapter. This might be unexpected to users, and unintended (not sure if all our adapter are on Jakarta EE yet).

@mposolda what is your take on this? Seems to me that this has been out for quite some time, so I don't see this as particularity harmful.

**Comment by mposolda** [comment]
> > Adding the dependency to the `common` module will add this dependency also to several adapters, for example in the Servlet adapter and the Tomcat adapter. This might be unexpected to users, and unintended (not sure if all our adapter are on Jakarta EE yet).
> 
> @mposolda what is your take on this? Seems to me that this has been out for quite some time, so I don't see this as particularity harmful.

This is good catch and I am sorry I did not spot this before. Just checked our adapters for OIDC and SAML https://www.keycloak.org/downloads . For example for Jetty, we support 9.4, which is not yet on Jakarta per https://projects.eclipse.org/projects/rt.jetty/releases/10.0 .

Perhaps we might revisit this later, but maybe it is possible to do something without introducing this new dependency to the `common` module? Like just make sure that the server stuff uses jakarta.

**Comment by pedroigor** [comment]
FYI, we have our own `Http` API to abstract how we manage requests and responses using the underlying stack. See https://github.com/keycloak/keycloak/tree/main/server-spi/src/main/java/org/keycloak/http.

This API was added to avoid direct dependencies on the underlying JAX-RS stack (e.g.: Resteasy) so that we make it easier to integrate with these different stacks. I forgot to mention this earlier but I think we should look at it as the right place to deal with cookies on the server-side.

However, the cookie-related code we have in `common` seems to be also used by some adapters. Perhaps that will mean reviewing how adapters are using this code and avoid sharing this code between the server and the adapter?

**Comment by jonkoops** [comment]
> FYI, we have our own `Http` API to abstract how we manage requests and responses using the underlying stack. See https://github.com/keycloak/keycloak/tree/main/server-spi/src/main/java/org/keycloak/http.

Am I confused then? It seems that this already has a [direct dependency](https://github.com/keycloak/keycloak/blob/ccade622898e3e14c2ff52f8bfeaa9d4d265a6a6/server-spi/src/main/java/org/keycloak/http/HttpCookie.java#L27) on JAX-RS.

**Comment by jonkoops** [comment]
Closing this as it was resolved under #26558

**Pull request closed without merging**